tensorflow/python/eager/forwardprop.py

```diff
@@ -439,3 +439,19 @@ def _fetch_jvp(tensor):
         return array_ops.zeros_like(tensor)
       return result
     return nest.map_structure(_fetch_jvp, primals)
+
+  @classmethod
+  def _batch_accumulator(cls, primals, tangents):
+    acc = super(ForwardAccumulator, cls).__new__(cls, primals, tangents)
+    acc._recording = False
+    acc._accumulator = pywrap_tfe.TFE_Py_ForwardAccumulatorNew(True)
+    primal_ids = set()
+    for primal in nest.flatten(primals):
```

Comment:
This might be a nice place to do some shape checking. Something like `tangent.shape.assert_is_compatible_with(tf.TensorShape(None) + primal.shape)`